ios/PolyPodApp/PodApi/PostOffice.swift

```diff
@@ -162,25 +173,39 @@ extension PostOffice {
     }
     
     private func handlePolyInDelete(args: [Any], completionHandler: @escaping (MessagePackValue?, MessagePackValue?) -> Void) {
-        let extendedDataSet = convertArgs(args: args, completionHandler: completionHandler)
+        let extendedDataSet: [ExtendedData]
+        do {
+            extendedDataSet = try convertArgs(args: args)
+        } catch {
+            completionHandler(nil, createErrorResponse(#function, error))
+            return
+        }
         
-        PodApi.shared.polyIn.deleteQuads(quads: extendedDataSet) { didDelete in
-            if didDelete {
-                completionHandler(MessagePackValue(), nil)
-            } else {
-                completionHandler(nil, MessagePackValue("Failed"))
+        PodApi.shared.polyIn.deleteQuads(quads: extendedDataSet) { error in
+            if let error = error {
+                completionHandler(nil, createErrorResponse(#function, error))
+                return
             }
+            completionHandler(MessagePackValue(), nil)
         }
     }
     
     private func handlePolyInHas(args: [Any], completionHandler: @escaping (MessagePackValue?, MessagePackValue?) -> Void) {
-        let extendedDataSet = convertArgs(args: args, completionHandler: completionHandler)
+        let extendedDataSet: [ExtendedData]
+        do {
+            extendedDataSet = try convertArgs(args: args)
+        } catch {
+            completionHandler(nil, createErrorResponse(#function, error))
+            return
+        }
         
         PodApi.shared.polyIn.hasQuads(quads: extendedDataSet) { doesHave in
             if doesHave {
                 completionHandler(MessagePackValue(), true)
```

Comment:
If we're not using the error, shouldn't this be nil?  Or simply nothing? Since this is an optional value and all...

Reply:
You're right, this is actually plain weird...